src/main/kotlin/rys/ajaxpetproject/service/impl/ChatServiceImpl.kt

```diff
@@ -0,0 +1,38 @@
+package rys.ajaxpetproject.service.impl
+
+
+import org.bson.types.ObjectId
+import org.springframework.stereotype.Service
+import rys.ajaxpetproject.exception.ChatNotFoundException
+import rys.ajaxpetproject.exception.ChatsNotFoundException
+import rys.ajaxpetproject.model.MongoChat
+import rys.ajaxpetproject.repository.ChatRepository
+import rys.ajaxpetproject.service.ChatService
+
+@Service
+class ChatServiceImpl(val chatRepository: ChatRepository) : ChatService {
+
+    override fun createChat(mongoChat: MongoChat) = chatRepository.save(mongoChat)
+
+    override fun findChatById(id: ObjectId): MongoChat? = chatRepository.findChatById(id)
+
+    fun getChatById(id: ObjectId) = chatRepository.findChatById(id) ?: throw ChatNotFoundException()
+
+    fun getAllChats(): List<MongoChat> = chatRepository.findAllBy() ?: throw ChatsNotFoundException()
+
+    override fun findAllChats(): List<MongoChat>? = chatRepository.findAllBy() ?: emptyList()
```

Comment:
will this method return nullable result?

Reply:
I consider that maybe there will be scenarios where the absence of chats is a normal situation. 
That's not the final method. These commits are more about how to write code. So I would know what code you expect from me.
For example, findAllChats() will be changed to findAllUserChats(), and in my opinion, if the user has no chats, it's normal behavior.

Reply:
the usual contract of findAll... methods is that they will return an empty list in case nothing is found. 
Moreover, in this particular case, you have logic that returns an empty list event if the underlying repository returns null. And clients of your service should handle null cases even if they can never happen

Reply:
So I should make non-nullable, remove ?: or remove whole method?